packages/expo-cli/src/commands/eject/Eject.js

```diff
@@ -101,241 +79,180 @@ export async function ejectAsync(projectRoot: string, options) {
+  if (!pkgJson.scripts) {
+    pkgJson.scripts = {};
+  }
+  delete pkgJson.scripts.eject;
+  pkgJson.scripts.start = 'react-native start';
+  pkgJson.scripts.ios = 'react-native run-ios';
+  pkgJson.scripts.android = 'react-native run-android';
+
+  const { sdkVersion } = exp;
+  const versions = await Versions.versionsAsync();
+  const reactNativeVersion = versions.sdkVersions[sdkVersion].facebookReactNativeVersion;
+  pkgJson.dependencies['react-native'] = reactNativeVersion;
 
-    await fse.writeFile(path.resolve('package.json'), JSON.stringify(pkgJson, null, 2));
+  // TODO: how should we version react-native-unimodules to match up with react-native version?
+  pkgJson.dependencies['react-native-unimodules'] = '^0.4.1';
 
-    log(chalk.green('Your package.json is up to date!'));
+  await fse.writeFile(path.resolve('package.json'), JSON.stringify(pkgJson, null, 2));
 
-    log(`Adding entry point...`);
-    const indexjs = `import { AppRegistry } from 'react-native';
+  log(chalk.green('Your package.json is up to date!'));
+
+  log(`Adding entry point...`);
+  delete pkgJson.main;
+
+  const indexjs = `import { AppRegistry } from 'react-native';
 import App from './App';
 
 AppRegistry.registerComponent('${appJson.name}', () => App);
 `;
-    await fse.writeFile(path.resolve('index.js'), indexjs);
-    log(chalk.green('Added new entry points!'));
+  await fse.writeFile(path.resolve('index.js'), indexjs);
+  log(chalk.green('Added new entry points!'));
 
-    log(`
+  log(`
 Note that using \`${npmOrYarn} start\` will now require you to run Xcode and/or
 Android Studio to build the native code for your project.`);
 
-    log('Removing node_modules...');
-    await fse.remove('node_modules');
-    if (useYarn) {
-      log('Installing packages with yarn...');
-      const args = newDevDependencies.length > 0 ? ['add', '--dev', ...newDevDependencies] : [];
-      spawn.sync('yarnpkg', args, { stdio: 'inherit' });
-    } else {
-      // npm prints the whole package tree to stdout unless we ignore it.
-      const stdio = [process.stdin, 'ignore', process.stderr];
-
-      log('Installing existing packages with npm...');
-      spawn.sync('npm', ['install'], { stdio });
-
-      if (newDevDependencies.length > 0) {
-        log('Installing new packages with npm...');
-        spawn.sync('npm', ['install', '--save-dev', ...newDevDependencies], {
-          stdio,
-        });
-      }
-    }
-  } else if (ejectMethod === 'expokit') {
-    await loginOrRegisterIfLoggedOut();
-    await Detach.detachAsync(projectRoot, options);
-  } else if (ejectMethod === 'cancel') {
-    // we don't want to print the survey for cancellations
-    log('OK! If you change your mind you can run this command again.');
-    return;
+  log('Removing node_modules...');
+  await fse.remove('node_modules');
+  if (useYarn) {
+    log('Installing packages with yarn...');
+    spawn.sync('yarnpkg', [], { stdio: 'inherit' });
```

Comment:
You could use this utility I made for `expo install`: [PackageManager](https://github.com/expo/expo-cli/blob/ed0bcca6952817666ca1931d23622b5aa2d5dd74/packages/expo-cli/src/PackageManager.js).

```js
const packageManager = PackageManager.createForProject(projectRoot);
await packageManager.installAsync();
```
It automatically uses npm or Yarn depending on the project and as a bonus filters out all the ugly peer dependency warnings.

And because npm and Yarn quite smart about purging removed packages, we might not have to delete `node_modules` anymore.